hs_core/hydroshare/hs_bagit.py

```diff
@@ -138,6 +138,16 @@ def create_bag_files(resource):
     for f in resFiles:
         a.add_resource(f)
 
+    # in current implementation, collection has no resource files
+    if resource.resource_type == "CollectionResource" and resource.metadata.collection:
+        for contained_res in resource.metadata.collection.resources.all():
+            contained_res_id = contained_res.short_id
+            landing_page = '{hs_url}/resource/{res_id}/'.format(hs_url=current_site_url, res_id=contained_res_id)
```

Comment:
@zhiyuli - Nearest I can tell looking at the Resource Map document and using the RDF validation service at https://www.w3.org/RDF/Validator/ - your resource map now looks correct.  You essentially have a resource map document for your collection that contains an "aggregation" and inside that aggregation are pointers to the resource map documents for the resources aggregated into the collection. If one wanted to - the resource map documents for the aggregated resources could be retrieved and interrogated to get the individual files identified within their respective maps (including their science metadata files). And - every file in every resource is identified via a resolvable URI.